src/geom/face_tri6.C

```diff
@@ -174,6 +174,17 @@ dof_id_type Tri6::key (const unsigned int s) const
 
 
 
+unsigned int Tri6::which_node_am_i(unsigned int side,
+                                   unsigned int side_node) const
+{
+  libmesh_assert_less (side, this->n_sides());
+  libmesh_assert_less (side_node, 3);
+
+  return Tri6::side_nodes_map[side][side_node];
+}
+
```

Comment:
The dozens of copies of essentially-identical code here make me sad inside.  I can't think of an obvious way to improve it, though.  Make a virtual function that returns side_nodes_map (and one that returns nodes_on_side(s), for the second assert), and then make Elem::which_node_am_i non-virtual?